src/gf.c

```diff
@@ -2028,6 +2028,17 @@ JL_DLLEXPORT void jl_method_table_disable(jl_methtable_t *mt, jl_method_t *metho
     JL_UNLOCK(&world_counter_lock);
 }
 
+JL_DLLEXPORT void jl_method_table_set_frozen(jl_methtable_t *mt, int val)
+{
+    JL_LOCK(&world_counter_lock);
+    JL_LOCK(&mt->writelock);
+    size_t world = jl_atomic_load_relaxed(&jl_world_counter);
+    mt->frozen = val;
+    jl_atomic_store_release(&jl_world_counter, world + 1);
+    JL_UNLOCK(&mt->writelock);
+    JL_UNLOCK(&world_counter_lock);
```

Comment:
The rest of the content doesn't make any sense. It is true this adds a possible data-race, but you'd need to change the field type to atomic to fix that. It also might make more sense to give this field a name (jl_methtable_type in jltypes.c) so that it can be manipulated from Julia, rather than adding a private accessor function and ccall just to manipulate the value?
```suggestion
    mt->frozen = val;
```